### PR TITLE
Add test for #2867

### PR DESCRIPTION
**Pull request by SimonHeimberg**

### CODE REVIEW

File: tests/Regression/GitHub/2867/Issue2861Test.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    public static function multipleIteratorProvider()
+    {
+        yield from self::iteratorProvider();
+        yield array(22, 2);
+        yield from self::arrayProvider();
+        yield from array(array(23, 31), array(24, 44));
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (SimonHeimberg) [reply tests/Regression/GitHub/2867/Issue2861Test.php]: agree.

File: tests/Regression/GitHub/2867/Issue2861Test.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+class Issue2867Test extends TestCase
```
[the remark this answers is not in the document]
Comment (SimonHeimberg) [reply tests/Regression/GitHub/2867/Issue2861Test.php]: What namespace? Is not done in similar places.

**Comment by SimonHeimberg** [comment]
using `phpunit --debug` to show the datasource names is just a suggestion. Alternatives: let tests fail, let them report skip and use `phpunit -v`

File: tests/Regression/GitHub/2867/Issue2867Test.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+class Issue2867Test extends TestCase
```
[the remark this answers is not in the document]
Comment (SimonHeimberg) [reply tests/Regression/GitHub/2867/Issue2867Test.php]: What namespace? (`use Regression;`, ...) Is not done in similar places.

[the remark this answers is not in the document]
Comment (sebastianbergmann) [reply tests/Regression/GitHub/2867/Issue2867Test.php]: You do realize that you're talking to a bot, right? :-)

[the remark this answers is not in the document]
Comment (SimonHeimberg) [reply tests/Regression/GitHub/2867/Issue2867Test.php]: Yes, I do. But I hoped that a human would answer and tell me if this is relevant.

[the remark this answers is not in the document]
Comment (sebastianbergmann) [reply tests/Regression/GitHub/2867/Issue2867Test.php]: It's not :)

File: tests/Regression/GitHub/2867-data-provider-keys.phpt [hunk 4]
```diff
@@ -0,0 +1,39 @@
+--TEST--
+GH-2867: test names show numeric datasource only for iterators as dataprovider
+GH-2639: Second `yield from` is not called from a data provider
+--FILE--
+<?php
+$_SERVER['argv'][1] = '--no-configuration';
+$_SERVER['argv'][1] = '--debug';
```
Comment (SimonHeimberg) [inline tests/Regression/GitHub/2867-data-provider-keys.phpt]: using `phpunit --debug` to show the datasource names is just a suggestion.
Alternatives:
 * let tests fail
 * let them report skip and use `phpunit -v`

**Comment by sebastianbergmann** [comment]
#2867 has been closed.

**Pull request closed without merging**